### ISSUE TITLE
Consensus on what the desired final output of the compactor

### ISSUE THREAD

**Issue by NGA-TRAN** [influxdata/influxdb_iox#5067]
# Rationale

This is a task of https://github.com/influxdata/conductor/issues/1023 to have consensus on what the desired final output of the compactor is and add it as comments in the main function of the `compact_partition`

This needs @pauldix to review:

The eventually desired output of a compacted partition is to include N non-overlapped files, each does not exceed a config param`desired_max_file_size_byte`.

We will adjust the setting `desired_max_file_size_byte` as we go to have:
  1. Good compaction performance. Which means
         i.  `Compact a partition` should finish in a reasonable time (TODO: quantify `reasonable time` in in seconds or minutes)
        ii. The Compactors catches up well with Ingesters (TODO: define what this means. For examples, if for some reason, the Compactors are down for several hours/days, when they come back on, how soon they bring the latest partitions to the desired output, how soon they bring the rest of the partitions to the desired output)
  2. Good query performance (TODO: define what good query performance means in the context of compaction such as max file size, number of non-overlapped files per partition, acceptance overlapped files or not) 

Note: the above consensus does not consider deletes that happens on cold partitions (aka partitions that have no ingesting data). This has lower priority and will be done in different (background) compaction cycle that needs new design.

Update: and we do hope N will be small (<= 10 or so)

**Comment by alamb** [comment]
I think it would help to define "over-lap" in this case to mean "non-overlapped time ranges". It also might help to clarify that this holds after some period of time in which the partition has not received any writes. 

Maybe it would also help to say that "min and max sequence numbers are the same", though perhaps if they aren't overlapped in time it doesn't matter. 

So maybe something like:

After a partition of a table has not received any writes for some amount of time, the compactor will ensure it is stored in object store as N parquet files files which:
1.  have non overlapping file ranges
2.  each does not exceed a size specified by config param `desired_max_file_size_byte.`


I would also suggest not describing how to tune `desired_max_file_size_byte` at this stage, especially since I think there may be many ways to improve compaction performance (e.g run more compactors, run more stuff in parallel, optimize datafusion plans, etc)

**Comment by Joe-Blount** [comment]
@alamb in your comment, should `1.` be:   _have non overlapping time ranges_

**Comment by alamb** [comment]
(fixed -- thank you @Joe-Blount )

**Comment by NGA-TRAN** [comment]
Thanks @alamb . I will us your version to add as comments in the code

**Comment by NGA-TRAN** [comment]
@jacobmarble 
> Do you mean that this issue is concerning compaction in hot partitions?

No, the deletes on hot partitions (when we turn delete on) will apply during compaction and it should be fine. What I mean here is our compaction only gets run on hot partition (aka partition with new writes). If deletes are issued for partitions without new writes, we will not do anything with it. This will lead to bad query performance on those cold partitions because. We have a plan to support compact cold partitions if there are deletes but apparently this is low priority

**Comment by jacobmarble** [comment]
Ah. I quoted you saying "deletes" and didn't realize it. Thank you.
